pkg/sessions/dynamodb/dynamodb_store.go

```diff
@@ -0,0 +1,139 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	dynamo "github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/sessions/persistence"
+)
+
+const SessionKeyName = "SessionKey"
+
+// GetPutDeleteWithContext is a subset of the dynamodbiface.DynamoDBAPI interface that defines only the operations
+// we need. This is useful for testing as it enables us to provide a custom client that doesn't have to implement the full interface.
+type GetPutDeleteWithContext interface {
```

Comment:
NIT: This feels like an odd name. You are already under the `dynamodb` package, so you can just call this `Client` (its calls would be `dynamodb.Client`).

Redis name the interface under its package `Client` -- I would lean toward just aligning with that.

Reply:
Plus 1 for calling this interface `Client`